README.md

```diff
@@ -207,23 +220,265 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
 | pyVersion        | Returns the version of Python                      | `pyVersion()`                                                        |
 
 
-# Usage Examples   
+# Wrapping python packages
+
+The following tools help generate R functions wrapping Python functions along with reference documentation (.Rd files) for the wrapper functions, the documentation being generated from the Sphinx-based Python doc-strings.
+
+Preconditions:
+* The Python package is downloaded and installed on the machine
+* The Python package is available on python search path
+
+A Python package may have multiple modules, each with its namespace. Each module has its own functions, classes, and variables. Each function or class has its own local namespace. In R, all functions and classes within the same package share a package namespace. To avoid the namespace collisions and allow customization to the wrapping package, we provide the following functions to help you pick which modules, functions, and classes to expose in R:
+
+* `generateRWrappers`
+* `generateRdFiles`
+
+The .Rd file generation must happen at the time the package is built, not at the time the package is loaded. The R function and contructor wrappers must happen at the time the package is loaded. These functions should be called with the similar params to ensure that all R wrapped functions have corresponding reference documentation.
+
+## Examples:
+
+In this example, I will demonstrate how we generate `synapser` and `synapserutils` packages by wrapping Python package `synapsePythonClient`.
+
+`synapsePythonClient` has the following structure:
+```
+synapsePythonClient package
+    module: synapseclient
+        module: client
+            class: Synapse
+                function: get
+                function: login
+        module: entity
+            class: Entity
+                function: get
+                function: privateGet
+            class: File
+                function: get
+                function: privateGet
+            class: Folder
+                function: get
+                function: privateGet
+        module: table
+            function: Table
+            function: column_ids
+            class: Column
+            class: CsvFileTable
+        module: cache
+            function: set
+    module: synapseutils
+        module: copy
+            function: copy
+            function: copyWiki
+            function: copyFileHandles
+        module: sync
+            function: syncToSynapse
+            function: syncFromSynapse
+        module: monitor
+            function: notifyMe
+```
+
+### Expose all functions and classes within a Python module
+
+For the R package `synapserutils`, our first attempt is to exposing all functions under `synapseutils` Python module. In `.configure` file, we execute an R script that calls `generateRdFiles` as following:
```

Comment:
To generate the R package `synapserutils` which wraps all functions in the `synapseutils` module, ...